starter_code/components.js

```diff
@@ -0,0 +1,32 @@
+class AwesomePictureThing extends React.Component {
+    
+    constructor(){
+        super();
+        this.state = {show: false}
+    }
+    _toggle(){
+        if(this.state.show === true) {
+            this.setState({show: false});
+        }
+        else { this.setState({show: true});}
+    }
+    render() {
+        if(this.state.show === true) {
```

Comment:
Again, is there a shorter way to do this that's more DRY? Rather than repeating the lines of code in JSX that are the same between both templates, could we just render one template and conditionally show or hide the image?

```js
return ( 
  <div>
    {this.state.show ? <img src="http://0.media.dorkly.cvcdn.com/77/37/c7358e161859b0a65c09fef0e3996c13.gif" /> : null }
    <button onClick={this._toggle.bind(this)}>buttON TO TOGGLE AHHHHHH</button>
  </div>
)
```

The [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) is very useful for these situations.